### PR TITLE
Tucana 20.7.0 plat 24535

### PR DESCRIPTION
**Pull request by danbar0kal**

### CODE REVIEW

File: api_v3/services/LiveStreamService.php [hunk 1]
```diff
@@ -11,6 +11,9 @@ class LiveStreamService extends KalturaLiveEntryService
 {
 	const ISLIVE_ACTION_CACHE_EXPIRY_WHEN_NOT_LIVE = 5;
 	const ISLIVE_ACTION_CACHE_EXPIRY_WHEN_LIVE = 30;
+
+	const GET_LIVE_STATS = 5;
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:15]: let's take this value from the live configuration map

File: api_v3/services/LiveStreamService.php [hunk 2]
```diff
@@ -617,6 +620,38 @@ public function archiveAction($liveEntryId, $vodEntryId)
         return true;
     }
 
+	private function responseHandlingGetStats($isLive)
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:623]: protected and not private

File: api_v3/services/LiveStreamService.php [hunk 3]
```diff
@@ -617,6 +620,38 @@ public function archiveAction($liveEntryId, $vodEntryId)
...
+	{
+		if ($isLive)
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:625]: i think that even if we are not live we need to have cache expiry

File: api_v3/services/LiveStreamService.php [hunk 4]
```diff
@@ -649,6 +684,18 @@ public function getDetailsAction($id)
 		return $liveStreamDetails;
 	}
 
+	protected function getNumberOfViewers($entryId)
+	{
+		$cache = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_PLAYS_VIEWS);
+		if (!$cache)
+		{
+			throw new kUploadTokenException("Cache instance required for Live Viewers Caching Could not initiated", kCoreException::PLAYS_VIEWS_CACHE_NOT_INITIALIZED);
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:692]: we usually dont throw such type of errors

File: api_v3/services/LiveStreamService.php [hunk 5]
```diff
@@ -649,6 +684,18 @@ public function getDetailsAction($id)
...
+		}
+		$numberOfViewers = $cache->get("plays_views_" . $entryId);
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:694]: why you are using plays_views key? you should have you own format of a key for the live viewers like "current_viewers_{entry_id}

File: api_v3/services/LiveStreamService.php [hunk 6]
```diff
@@ -649,6 +684,18 @@ public function getDetailsAction($id)
...
+
+		return $numberOfViewers ?: 0;
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:696]: not clear

File: api_v3/services/LiveStreamService.php [hunk 7]
```diff
@@ -617,6 +618,46 @@ public function archiveAction($liveEntryId, $vodEntryId)
         return true;
     }
 
+	protected function responseHandlingGetStats($isLive)
+	{
+		$liveStatsInterval = kConf::get('liveStatsInterval','live');
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:623]: take live mape name from kConfMapNames class
also add default value of 5 for the interval

File: api_v3/services/LiveStreamService.php [hunk 8]
```diff
@@ -617,6 +618,46 @@ public function archiveAction($liveEntryId, $vodEntryId)
...
+	 * @ksIgnored
+	 *
+	 * @throws KalturaErrors::INVALID_ENTRY_ID
+	 */
+	public function getLiveStreamStatsAction($entryId)
+	{
+		$liveStreamEntry = $this->fetchLiveEntry($entryId);
+		$isLive = $liveStreamEntry->isCurrentlyLive();
+		$liveStreamInfo = new KalturaLiveStreamStats();
+		$liveStreamInfo->liveViewers = $isLive ? $this->getNumberOfViewers($entryId) : 0;
```
Comment (inbal-ben-david) [inline api_v3/services/LiveStreamService.php:643]: lets avoid extracting the entry and just return what we have in cache, if its not live the value will be null

File: api_v3/services/LiveStreamService.php (hunk 7 above)
Comment (danbar0kal) [reply api_v3/services/LiveStreamService.php:623]: didnt nadav asked for 20?

Comment (inbal-ben-david) [reply api_v3/services/LiveStreamService.php:623]: yes it was before, put 20

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**